### ISSUE TITLE
Table Scrollbars not always usable

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [grafana/grafana#52136]
**What happened**:

When using Grafana Explore and results are in table format, vertical and horizontal scrollbars of the table may be hard to use because they are outside the visible area (see below).

Depending on browser / OS settings (?), the barely visible scrollbars may even fade out and become completely invisible, resulting in a frustrating experience (See video especially).

May also be an UX design issue / Accessibility issue?

**What you expected to happen**:

Table scrollbars should always be accessible.

**How to reproduce it (as minimally and precisely as possible)**:

- Get a result table with some data in Explore, e.g. use this link: https://play.grafana.org/explore?orgId=1&left=%7B%22datasource%22:%22Prometheus:%20Robust%20Perception%20Sample%20Endpoint%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22topk(100,%20%7Bjob%3D~%5C%22.%2B%5C%22%7D)%22,%22range%22:false,%22editorMode%22:%22code%22,%22instant%22:true,%22exemplar%22:false%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D
- The table has several columns and rows, requiring scrollbars in both directions

### Issue 1) Vertical Scrollbar not accessible

The table has to be scrolled to the far right to get to the vertical scrollbar

![grafik](https://user-images.githubusercontent.com/33002073/178686316-7844cc3c-13f0-4f92-80c2-1a665e71202c.png)

GIF:

![explorer_jKmcWk1haH](https://user-images.githubusercontent.com/33002073/178687140-70a7d962-eeeb-4755-bbe7-4fde21c0551a.gif)

### Issue 2) Horizontal Scrollbar not accessible

The table does seem to have a preferred height, which may lead to the horizontal scrollbar also not being accessible.
The whole explore pages gets a vertical scrollbar, which is barely visible at all.

Using the mouse wheel may not always help, because if hovering on the data, the table gets scrolled, not the explore page.

OK if enough vertical space:

![grafik](https://user-images.githubusercontent.com/33002073/178687872-41880815-89ec-4ce8-8bc3-bd2a71522365.png)

After expanding panels NOK:

![grafik](https://user-images.githubusercontent.com/33002073/178688370-38f170c4-9f39-4620-a0a4-a6a6988f99e8.png)

Video:

https://user-images.githubusercontent.com/33002073/178688784-3a8a58f1-d2a2-47f2-a238-83fe4efb50a2.mp4

**Anything else we need to know?**:

Screenshots are from browsers running at 1920x1080, 100% zoom.

Not sure if this issue also applies to dashboards or only explore.

**Environment**:
- Grafana version: `8.5.0` and `9.0.3`
- User OS & Browser: Windows, Firefox `102.0.1` and Google Chrome `103`

**Comment by codeincarnate** [comment]
If I remember correctly there's difficulty in implementing this because of react table usage and needing to track scroll state between a scrollbar at the end and one that floats above the table. We're looking into this but it may take some time.

**Comment by mdvictor** [comment]
Hello,

The table panel vertical scroll issue will be fixed by https://github.com/grafana/grafana/pull/53457

However, @Mario-Hofstaetter, I'm having trouble understanding the second issue you propose. The table horizontal bar is accessible if you scroll the explore page down. It works both with mouse click and on mouse scroll (as long as you are in the context of the explore page).

Are you suggesting an issue in functionality or more of a UX issue where the bar is not visible very well?

**Comment by Mario-Hofstaetter** [comment]
> The table horizontal bar is accessible if you scroll the explore page down. It works both with mouse click and on mouse scroll (as long as you are in the context of the explore page).
> Are you suggesting an issue in functionality or more of a UX issue where the bar is not visible very well?

@mdvictor I am suggesting an UX issue. IMHO it should not be required to scroll down the explore page itself, to get to the "bottom" of the table to reach its horizontal scrollbar.

Always when having multiple vertical scrollbars, behavior becomes confusing from user perspective, e.g. Mouse Scrolling is depending on the hover position of the cursor, especially with modern style slim / hard to see scrollbars and litte contrast between layout sections...

If an element has a fixed / perferred height, e.g. a CHART or a JPEG with x*y pixels, it is obvious what its height is (should be).

But a table is scrollable and dynamic itself. On a dashboard, the user places the table panel and sets its desired height (in cells of the dashboard), but in Explore the layout is fixed.

So in my opinion, the table in explore should "fit" into the available vertical space in explore, depending on other elements. If however space is very tight, this may not be possible (hence a minimum preferred size).

I don't know if this makes sense... I'll try with an example.

Using [the link from above](https://play.grafana.org/explore?orgId=1&left=%7B%22datasource%22:%22Prometheus:%20Robust%20Perception%20Sample%20Endpoint%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22topk(100,%20%7Bjob%3D~%5C%22.%2B%5C%22%7D)%22,%22range%22:false,%22editorMode%22:%22code%22,%22instant%22:true,%22exemplar%22:false%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D) and expanding `Options`, the horizontal scrollbar of the table is not accessible.

This is not plausible for a common user. The browser windows is at 100%, with 1920x1080px.
The table has a height of ~ 630px, one would expect to see the table horizontal scrollbar at the bottom.

Instead, the explore page has a vertical scrollbar. Using the mousewheel is not possible if the mouse is above the table (which is reasonable) and the table has many rows, because the table is scrolled, not the explore page.

If you scroll to the end of the table data, the explore page is eventually scrolled down. But the table may contain 100.000 rows, so you would scroll quite some time.

![grafana-explore-scroll](https://user-images.githubusercontent.com/33002073/183646744-95db1b7e-bb96-4063-8bce-506afffcfd10.png)

When opening the explore page initially, before expanding "Options", the table horizontal scrollbar is still visible, but the Explore page has already a minimal vertical bar. Video showing this all together:

https://user-images.githubusercontent.com/33002073/183647341-2e81f28e-fdb8-40e5-a315-470c6f0e34bf.mp4

For this layout at this resolution, the explore page should not have a vertical scrollbar.
The table height should use the available space left by panels above (query editor, Options, metrics browser).

If space is little, the vertical scrollbar is OK, for example when using multiple queries or a smaller browser window:

![little space in explore](https://user-images.githubusercontent.com/33002073/183648105-1c731ed7-8eb2-407d-8d5a-eaceab8e63df.png)

**Comment by mdvictor** [comment]
@ashharrison90 can you please provide and answer here? I am afraid I don't know what to answer regarding explore and the table panel fitting the window under certain circumstances as suggested.

**Comment by mdvictor** [comment]
I am going to close this with the mention that the issue has been fixed and the scrollbars are always usable on a panel. There is the case in Explore where the table is larger than the viewport, in which case the entire viewport can be scrolled to allow table scrollbars to be used.
